### PR TITLE
fix: noImplicitAny: true

### PR DESCRIPTION
**Pull request by UziTech**
## Description

Remove `noImplicitAny: false` from tsconfig

## Contributor

- [x] Test(s) exist to ensure functionality and minimize regression (if no tests added, list tests covering this PR); or,
- [ ] no tests required for this PR.
- [ ] If submitting new feature, it has been documented in the appropriate places.

## Committer

In most cases, this should be a different person than the contributor.

- [ ] CI is green (no forced merge required).
- [ ] Squash and Merge PR following [conventional commit guidelines](https://www.conventionalcommits.org/).

### CODE REVIEW

File: lib/marked.cjs [hunk 1]
```diff
@@ -2024,23 +2027,27 @@ var Marked = class {
       if (pack.renderer) {
         const renderer = this.defaults.renderer || new _Renderer(this.defaults);
         for (const prop in pack.renderer) {
-          const prevRenderer = renderer[prop];
-          renderer[prop] = (...args2) => {
-            let ret = pack.renderer[prop].apply(renderer, args2);
+          const rendererFunc = pack.renderer[prop];
+          const rendererKey = prop;
```
Comment (styfle) [inline lib/marked.cjs:2031]: Is the variable for rendererKey needed? Can you use prop instead?

Comment (UziTech) [reply lib/marked.cjs:2031]: This is so Typescript knows what type prop is. Without this it is a string and renderer[prop] is `any` (implicitly) This is a generated file. in the .ts file it is `const rendererKey = prop as  keyof _Renderer;`

Comment (styfle) [reply lib/marked.cjs:2031]: Oops, I keep looking at the wrong files.

I think there is a way to tell github which files are generated so they start out collapsed in code reviews 🕵️‍♂️

Comment (styfle) [reply lib/marked.cjs:2031]: Added in PR #2908

File: src/Hooks.ts [hunk 2]
```diff
@@ -16,14 +16,14 @@ export class _Hooks {
   /**
    * Process markdown before marked
    */
-  preprocess(markdown: string) {
+  preprocess(markdown: string): string | Promise<string> {
```
Comment (styfle) [inline src/Hooks.ts:19]: Why `Promise<string>` here if its always returning a string?

Comment (UziTech) [reply src/Hooks.ts]: good catch! This was an early change

File: src/Instance.ts [hunk 3]
```diff
@@ -15,6 +15,9 @@ import type { Token, TokensList } from './Tokens.ts';
 
 export type ResultCallback = (error: Error | null, parseResult?: string) => undefined | void;
 
+type unknownFunction = (...args: unknown[]) => unknown;
```
Comment (styfle) [inline src/Instance.ts:18]:
```suggestion
type UnknownFunction = (...args: unknown[]) => unknown;
```

Lets use the same convention for all types (PascalCase)

File: src/Instance.ts [hunk 4]
```diff
@@ -140,25 +144,29 @@ export class Marked {
...
+            return ret || '';
           };
         }
         opts.renderer = renderer;
       }
       if (pack.tokenizer) {
         const tokenizer = this.defaults.tokenizer || new _Tokenizer(this.defaults);
         for (const prop in pack.tokenizer) {
-          const prevTokenizer = tokenizer[prop];
+          const tokenizerFunc = pack.tokenizer[prop as keyof MarkedExtension['tokenizer']] as unknownFunction;
```
Comment (styfle) [inline src/Instance.ts:164]:
```suggestion
          const tokenizerFunc = pack.tokenizer[prop as keyof MarkedExtension['tokenizer']] as UnknownFunction;
```

File: src/MarkedOptions.ts [hunk 5]
```diff
@@ -98,8 +100,8 @@ export interface MarkedExtension {
    * postprocess is called to process html after marked has finished parsing.
    */
   hooks?: {
-    preprocess: (markdown: string) => string,
-    postprocess: (html: string | undefined) => string | undefined,
+    preprocess: (markdown: string) => string | Promise<string>,
+    postprocess: (html: string) => string | Promise<string>,
```
[the remark this answers is not in the document]
Comment (UziTech) [reply src/MarkedOptions.ts:104]: The hooks can return promises if async option is true

[the remark this answers is not in the document]
Comment (styfle) [reply src/MarkedOptions.ts:104]: Hmm, then we might need to overload the types in order to avoid all the `as string` in `src/Instance.ts`

**Review by styfle — APPROVED** [review]

**Comment by UziTech** [comment]
@calculuschild any issue with merging this PR?

**Review by calculuschild — APPROVED** [review]
This is fine.

**Pull request merged**